docs/IAS-XSUAA-token-xchange.md

```diff
@@ -0,0 +1,68 @@
+In case token is not Xsuaa token, application checks if token exchange is enabled, if it is, then Xsuaa access token is requested from Xsuaa instance using the POST request with Header `X-zid`:`zone_uuid`, `grant_type` = `jwt-bearer` and `assertion` = IAS token from the request. Upon successful token exchange between IAS and XSUAA, token gets validated and depending on validation result authorized or unauthorized response is sent back.
+In situation when incoming request doesn't contain Xsuaa token and token exchange is disabled, unauthorized response is sent back. 
+
+## Enable token exchange in the application
+Set the environment variable `IAS_XSUAA_XCHANGE_ENABLED` to any value, but false or empty. For detailed token exchange setup information please see:
+- [spring-xsuaa readme](https://github.com/SAP/cloud-security-xsuaa-integration/tree/master/spring-xsuaa#ias-to-xsuaa-token-exchange) for Spring applications
+- [java-security readme](https://github.com/SAP/cloud-security-xsuaa-integration/tree/master/java-security#ias-to-xsuaa-token-exchange) for Java applications
+
+## Test the setup
+#### 1. Setup trust between IAS and Xsuaa
+The trust can be setup via
+[SCP Cockpit](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/161f8f0cfac64c4fa2d973bc5f08a894.html)
+
+#### 2. Create IAS service broker instance
+Create an instance of the IAS service broker
+
+`cf create-service identity application <IAS_SERVICE_INSTANCE_NAME> -c '{"xsuaa-cross-consumption": "true"}'`
+
+> The flag adds the client Id of the trusted XSUAA service, to the audience field of the ID token for cross consumption.
+
+#### 3. Fetch the IAS token
+- Create service key for IAS instance if it doesn't exist: `cf create-service-key <IAS_SERVICE_INSTANCE_NAME> <SERVICE_KEY_NAME>`
+- Fetch the client credentials: `cf service-key <IAS_SERVICE_INSTANCE_NAME> <SERVICE_KEY_NAME>`
+- Request a token:
+   ```shell script
+   curl POST 'https://iasTenant.accounts400.ondemand.com/oauth2/token' \
+   --header 'Authorization: Basic <IAS_CLIENT_ID> <IAS_CLIENT_SECRET>' \
+   --header 'Content-Type: application/x-www-form-urlencoded' \
+   --data-urlencode 'grant_type=password' \
+   --data-urlencode 'username=<YOUR_USERNAME>' \
+   --data-urlencode 'response_type=id_token' \
+   --data-urlencode 'password=<YOUR_IAS_PSWD>'
+   ```
+#### 4. Call secured endpoint with an IAS token
+Call the secured endpoint with the IAS token like in the example below
+   ```shell script
+    curl GET 'https://yourApp.cfapps.sap.hana.ondemand.com/yourSecuredEndpoint' \
+    --header 'Authorization: Bearer <FETCHED IAS TOKEN>'
+   ```
+You should receive an authorized response, if everything works fine.
+
+:interrobang: In case of 403 error, check if the user has required role assigned. 
```

Comment:
Hi @liga-oz does the role have to be assigned to "sap.custom" or "sap.default" origin or both?

Reply:
Hi @nenaraab, role has to be assigned to IAS tenant that has established trust with the XSUAA i.e. the IAS instance that has application in the space where Xsuaa instance and application were created. In your instance it's sap.custom origin, but the origin naming can vary, based on service setup.